world environment
position current_position
---
bool collision
